drivers/watchdog/Kconfig.nrfx

```diff
@@ -11,15 +11,21 @@ config WDT_NRFX
 	help
 	  Enable support for nrfx WDT driver for nRF MCU series.
 
+# TODO: bouncing this information through Kconfig is not really necessary.
+#
+# This is only used by nrfx_config.h in the Nordic HAL to convert
+# Kconfig to NRFX internal macros, but nrfx_config.h could use
+# devicetree.h directly instead.
+
+# Workaround for not being able to have commas in macro arguments
+DT_COMPAT_NORDIC_NRF_WATCHDOG  := nordic,nrf-watchdog
+
 config NRFX_WDT0
-	bool "Enable WDT0 instance"
-	depends on HAS_HW_NRF_WDT || HAS_HW_NRF_WDT0
-	default y
+	def_bool $(dt_nodelabel_has_compat,wdt0,$(DT_COMPAT_NORDIC_NRF_WATCHDOG))
```

Comment:
This option (as well as `NRFX_WDT1`) is [defined in modules/Kconfig.nordic](https://github.com/zephyrproject-rtos/zephyr/blob/2596fac9b72e1037e360c6cf06a1a8e155056495/modules/Kconfig.nordic#L444), so I think we should remove this second definition here and instead use `select NRFX_WDT0 if ...` in `config WDT_NRFX` above (`select NRFX_WDT` which is present there is actually redundant).

Reply:
I'm dropping all Kconfig changes until https://github.com/zephyrproject-rtos/zephyr/issues/10621 is resolved.